src/Amphitheatre/amphis.jl

```diff
@@ -0,0 +1,237 @@
+			push!(basicfg.nodes, vsym=>nodestruct)
+		else
+			basicfg.nodes[vsym] = nodestruct
+		end
+
+		visNode!(vis[robotId][sessionId][groupsym][vsym], nodestruct, isnewnode,
+				 zoffset = basicfg.zoffset,
+				 scale = drawProp.scale,
+				 sphereScale = drawProp.sphereScale,
+				 color = drawProp.color)
+
+		#FIXME this is bad, but just testing feature TODO figure out how to do it properly
+		basicfg.drawPath && groupsym == :poses && push!(trackPoints, Point3f0(xmx[1],xmx[2],0.0))
+
+    end
+
+	basicfg.drawPath && setobject!(vis[robotId][sessionId][:track], Object(PointCloud(trackPoints), LineBasicMaterial(), "Line"))
+
+    return nothing
+end
+
+
+# ============================================================
+# -----------------------BasicGraffPose-----------------------
+# ============================================================
+
+struct BasicGraffPose <: AbstractAmphitheatre
+	robotId::String
+	sessionId::String
+	config::GraffConfig
+	nodes::Dict{Symbol, AbstractPointPose} #poseId, AbstractPointPose
+
+	meanmax::Symbol
+	zoffset::Float64
+	drawPath::Bool
+	#pose drawing propeties
+	poseProp::plDrawProp
+	#landmark drawing propeties
+	landmarkProp::plDrawProp
+end
+
+"""
+   $(SIGNATURES)
+Basic visualizer object to draw poses and landmarks from GraffSDK.
+"""
+function BasicGraffPose(config::GraffConfig;
+							  meanmax::Symbol=:max,
+						      zoffset::Float64=0.0,
+							  drawPath::Bool=false,
+							  poseProp::plDrawProp = plDrawProp(0.1, 0.05, RGBA(1,1,0,0.5)),
+							  landmarkProp::plDrawProp = plDrawProp(0.2, 0.1, RGBA(0,1,0,0.5)))
+
+   return BasicGraffPose(config.robotId, config.sessionId, config, Dict{Symbol,AbstractPointPose}(),
+							    meanmax,
+							    zoffset,
+							    drawPath,
+							    poseProp,
+							    landmarkProp)
+end
+
+function visualize!(vis::Visualizer, grafffg::BasicGraffPose)::Nothing
+	#TODO maybe improve this function to lower memmory allocations
+
+
+	# get the Graff factor graph object
+	robotId = grafffg.robotId
+	sessionId = grafffg.sessionId
+
+	nodes = GraffSDK.getNodes(robotId, sessionId).nodes
+
+	for nod in nodes
+
+		if nod.mapEst == nothing
+		    continue
+		end
+		# get the variable type
+		nodedetail = getNode(nod.id)
+		typestr = split(nodedetail.type, ".")[end]
+		typesym = Symbol("Arena$typestr")
+
+		nodef = getfield(Amphitheatre, typesym)
+
+		#NOTE make sure storage order and softtypes are always the same
+		nodestruct = nodef(nod.mapEst...)
+
+
+		vsym = Symbol(nod.label)
+
+		nodelabels = nodedetail.labels
+		if in("LANDMARK", nodelabels)
+			groupsym = :landmarks
+			drawProp = grafffg.landmarkProp
+
+		elseif in("POSE", nodelabels)
+			groupsym = :poses
+			drawProp = grafffg.poseProp
+
+		elseif string(vsym)[1] == 'l'
+			groupsym = :landmarks
+			drawProp = grafffg.landmarkProp
+
+		elseif string(vsym)[1] == 'x'
+			groupsym = :poses
+			drawProp = grafffg.poseProp
+
+		else
+			@warn "Unknown symbol encountered $vsym"
+			groupsym = :unknown
+		end
```

Comment:
in future we can also break the "longer" ifelse statements into a separate function -- also not important now.

Reply:
noted